File: C++/program29.cpp

/*
    Input : 10
    Output: 2 
*/
#include<iostream>
using namespace std;

void DisplayEvenFactor(int iNo)
{
    if(iNo < 0)
    {
        iNo = -iNo;
    }
    int iCnt = 0;

    for(iCnt = 2; iCnt <= (iNo / 2); iCnt++)
    {
        if(((iNo % iCnt) == 0) && ((iCnt % 2) == 0))
        {
            cout<<"Even factor is  : "<<iCnt<<'\n';
        }
       
    }
}
int main()
{
    int iValue = 0;

    cout<<"Enter number"<<'\n';
    cin>>iValue;

    DisplayEvenFactor(iValue);

    return 0;
}

File: Python/program29.py
def main():
    
    iValue = int(input("Enter number \n"))
    
    iRet = Factorial(iValue)
    
    if (iRet == -1):
        print("Invalid Input\n")
    else:
        print("Factorial  is ", iRet)
    
def Factorial(iNo):
    Fact = 1
    if(iNo < 0):
        return -1
    
    for iCnt in range(1,iNo + 1):
        Fact = Fact * iCnt
    return Fact
    

    
    
if __name__ == "__main__":
    main()